File: Examples/test-suite/go/typedef_funcptr_runme.go
package main

import . "swigtests/typedef_funcptr"

func main() {
	a := 100
	b := 10

	if Do_op(a,b,Addf) != 110 {
		panic(0)
	}
	if Do_op(a,b,Subf) != 90 {
		panic(0)
	}

	if Do_op_typedef_int(a,b,Addf) != 110 {
		panic(0)
	}
	if Do_op_typedef_int(a,b,Subf) != 90 {
		panic(0)
	}

	if Do_op_typedef_Integer(a,b,Addf) != 110 {
		panic(0)
	}
	if Do_op_typedef_Integer(a,b,Subf) != 90 {
		panic(0)
	}
}

File: Examples/test-suite/lua/typedef_funcptr_runme.lua
local v=require("typedef_funcptr")

local a = 100
local b = 10
assert(v.do_op(a,b,typedef_funcptr.addf) == 110, "addf")
assert(v.do_op(a,b,typedef_funcptr.subf) == 90, "subf")
assert(v.do_op_typedef_int(a,b,typedef_funcptr.addf) == 110, "addf")
assert(v.do_op_typedef_int(a,b,typedef_funcptr.subf) == 90, "subf")
assert(v.do_op_typedef_Integer(a,b,typedef_funcptr.addf) == 110, "addf")
assert(v.do_op_typedef_Integer(a,b,typedef_funcptr.subf) == 90, "subf")
